pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -69,7 +70,7 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID string) *
 		meshExternal = true
 	}
 
-	if addr == constants.UnspecifiedIP && external == "" { // headless services should not be load balanced
+	if features.EnableHeadlessServicePassthrough.Get() && addr == constants.UnspecifiedIP && external == "" { // headless services should not be load balanced
```

Comment:
undo all changes here. Edit the code in cluster.go as you pointed out in slack
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster.go#L695
change function definition to take in the service object.
Then when processing passthrough type, check if feature enabled, and service.attributes.registry==kubernetes and if so, return EDS as the response.